src/test/scala/io/hydrosphere/mist/master/interfaces/http/HttpApiV2Spec.scala

```diff
@@ -40,6 +40,30 @@ class HttpApiV2Spec extends FunSpec with Matchers with ScalatestRouteTest {
       }
     }
 
+    it("should return bad request on futures failed illegal argument exception") {
+      val master = mock(classOf[MasterService])
+      val api = new HttpApiV2(master, mappings).route
+
+      when(master.runJob(any(classOf[JobStartRequest]), any(classOf[Source]), any[Action]))
+        .thenReturn(Future.failed(new IllegalArgumentException("argument missing")))
+
+      Post(s"/v2/api/endpoints/x/jobs", Map("1" -> "Hello")) ~> api ~> check {
+        status === StatusCodes.BadRequest
```

Comment:
Funny thing about `===` - it does not check anything. It was my copy-paste mistake and I fixed that everywhere in another branch.
Please, replace it with `shouldBe` in your tests